Core/include/Acts/EventData/SubspaceHelpers.hpp

```diff
@@ -0,0 +1,165 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/EventData/Types.hpp"
+#include "Acts/Utilities/AlgebraHelpers.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+#include <bitset>
+#include <span>
+
+namespace Acts {
+
+template <std::size_t kFullSize>
+class SubspaceHelper {
+ public:
+  explicit SubspaceHelper(std::span<const std::uint8_t> indices)
+      : SubspaceHelper(indices, indices.size()) {}
+
+  SubspaceHelper(std::span<const std::uint8_t> indices, std::size_t size)
```

Comment:
Is this so you can clip the input span?

Reply:
Yeah exactly - maybe I change this again now that I see it. I think it is okay to ask the user to provide the correct size.

Reply:
Maybe, I have a biased view, but I would say that it would be better if it was fixed sized i.e. a 
````
template <size_t N> ...(...array<uint8_t, N> ...)
```
that should allow for better optimizations. There should only be a couple of cases likely (N=1,2,3), and likely optimizations are better if these cases are handled explicitly. One could add the alternative to use a span, but there should be an option for a fixed size container.

Reply:
that might be a good option yes. I think the projector is only set in paths where we know the measurement dimension

Reply:
or fixed size span